plugins/gfonts.koplugin/api.lua

```diff
@@ -0,0 +1,89 @@
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local ltn12 = require("ltn12")
+local https = require("ssl.https")
+local rapidjson = require("rapidjson")
+local socket = require("socket")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local Api = {
+    base_url = "https://www.googleapis.com/webfonts/v1/webfonts",
+    font_variant = {
+        ["regular"] = "regular",
+        ["italic"] = "italic",
+        ["500"] = "medium",
+        ["500italic"] = "medium-italic",
+        ["700"] = "bold",
+        ["700italic"] = "bold-italic",
+    },
+}
+
+function Api:init(file)
+    local fallback_key = "AIzaSyDQZaihK8Lb7jJ3DYyrQrpyyJF7tyvrqAs"
+    if not file then
+        self.api_key = fallback_key
+        return
+    end
+    local keyfile = io.open(file, "r")
+    if keyfile then
+        local key = keyfile:read("*a")
+        keyfile:close()
+        self.api_key = key
+        return
+    end
+    self.api_key = fallback_key
+end
+
+function Api:request()
+    local request, sink = {}, {}
+    request['url'] = self.base_url .. "?key=" .. self.api_key
```

Comment:
No real reason to use request["url"] in places like this. (Unless you really want to I suppose.)
```suggestion
    request.url = self.base_url .. "?key=" .. self.api_key
```